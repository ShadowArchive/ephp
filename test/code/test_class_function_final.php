<?php

class A {
    final public function name($data) { return "Mike"; }
    public final function phone($data) { return "+1800"; }
}

class B extends A {
    public function name() { return "Roger"; }
    public function phone() { return "+0049"; }
}

$b = new B;
print $b->name() . "\n";

