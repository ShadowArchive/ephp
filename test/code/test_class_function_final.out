
Fatal error: Cannot override final method A::name() in {{CWD}}/test/code/test_class_function_final.php on line 8
